src/app/SubscriptionResumptionHelper.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/AttributePathParams.h>
+#include <app/CASESessionManager.h>
+#include <app/SubscriptionResumptionStorage.h>
+
+namespace chip {
+namespace app {
+class InteractionModelEngine;
+class ReadHandler;
+
+class SubscriptionResumptionHelper
+{
+public:
+    SubscriptionResumptionHelper();
+
+    ~SubscriptionResumptionHelper() {}
+
+private:
+    friend class InteractionModelEngine;
+    friend class ReadHandler;
+
+    CHIP_ERROR ResumeSubscription(CASESessionManager & caseSessionManager,
+                                  SubscriptionResumptionStorage::SubscriptionInfo & subscriptionInfo);
+
+    // Callback funstions for continuing the subscription resumption
+    static void HandleDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr,
+                                      const SessionHandle & sessionHandle);
+    static void HandleDeviceConnectionFailure(void * context, const ScopedNodeId & peerId, CHIP_ERROR error);
+
+    NodeId mNodeId;
```

Comment:
Why copy everything individually and not just use a `SubscriptionInfo` here?

Reply:
Make this class privately inherits from `SubscriptionInfo` now